go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kubebuilder/v3
 
-go 1.17
+go 1.18
```

Comment:
@oscr if we bump the Kubebuilder version from 0.17 that would go to 1.19 not 1.18
Kubebuilder is also used as LIB so that can impact those that consume the project.
Therefore, the bump go version would make sense to be done in a PR where we do only this change.
Note that we use the title and emojis to generate the changelogs.

Reply:
@camilamacedo86 Since we're starting to use go 1.19 should I upgrade everywhere?

Reply:
@camilamacedo86 Thank you for the advice. I will have it done in another pr.

Reply:
a) Do we need to change the go version to bump the declarative one? If so, can we bump the go version first in another PR?

b) This pr is not updating the controller-runtime see that it is done here: 
https://github.com/kubernetes-sigs/kubebuilder/blob/c7d38ce80602d4f2915a17f6f681a27901698de6/pkg/plugins/golang/v3/scaffolds/init.go#L35-L42. ()

**We can bump only the declarative one and do the other bumps in another PR that is totally fine.**
If you want to do that, just revert the golang changes and I am happy to lgtm/approve this one to get merged 🚀  

Also, you can check an example of a PR where we bump all deps, see: https://github.com/kubernetes-sigs/kubebuilder/pull/2471

Again, thank you for all your collab 🥇